uaclient/system.py

```diff
@@ -664,8 +663,8 @@ def _subp(
         except UnboundLocalError:
             raise exceptions.ProcessExecutionError(cmd=redacted_cmd)
 
-    out_result = out.decode("utf-8") if out else ""
-    err_result = err.decode("utf-8") if err else ""
+    out_result = out.decode("utf-8", errors="ignore") if out else ""
+    err_result = err.decode("utf-8", errors="ignore") if err else ""
```

Comment:
There is another decode call on the `snap` module. Let's add the same logic there too